core/src/main/java/org/bptlab/cepta/operators/EnrichDelayWithCoordinatesFunction.java

```diff
@@ -0,0 +1,97 @@
+    /* This Function takes an inputstream of DelayNotifications and enriches the events
+       with information about the coordinations of the dedicated station */
+
+    public EnrichDelayWithCoordinatesFunction(MongoConfig mongoConfig){
+        this.mongoConfig = mongoConfig;
+        this.tableName = "eletastations";
+    }
+
+    public EnrichDelayWithCoordinatesFunction(String tableName, MongoConfig mongoConfig){
+        this.mongoConfig = mongoConfig;
+        this.tableName = tableName;
+    }
+
+    private Hashtable<String, CoordinateOuterClass.Coordinate> coordinateMapping =  new Hashtable<>();
+
+    public Hashtable<String, CoordinateOuterClass.Coordinate> getMapping(){ return coordinateMapping;}
+
+    private MongoConfig mongoConfig = new MongoConfig();
+    private transient MongoClient mongoClient;
+    private String tableName;
+
+    private boolean readInStationData(){
+        MongoDatabase database = mongoClient.getDatabase("replay"/*mongoConfig.getName()*/);
+        MongoCollection<Document> eletastations = database.getCollection(tableName);
+
+        SubscriberHelpers.OperationSubscriber<Document> findMultipleSubscriber = new SubscriberHelpers.OperationSubscriber<>();
+        eletastations.find().subscribe(findMultipleSubscriber);
+
+        List<Document> allStations = findMultipleSubscriber.get();
+
+        for (Document station: allStations){
+            CoordinateOuterClass.Coordinate coordinate =
+                    CoordinateOuterClass.Coordinate.newBuilder()
+                            .setLongitude((double) station.get("longitude"))
+                            .setLatitude((double) station.get("latitude"))
+                            .build();
+            Long key = (Long) station.get("stationId");
+            coordinateMapping.put(String.valueOf(key), coordinate);
+        }
+        return true;
+    }
+
+
+    /**
+     * First we read in the station data into our local map variable.
+     * @param parameters
+     * @throws Exception
+     */
+    public void open(org.apache.flink.configuration.Configuration parameters) throws Exception {
+        super.open(parameters);
+        //TODO Remove Port after Demo!
+        this.mongoClient = Mongo.getMongoClient(mongoConfig/*.withPort(27017)*/);
+        readInStationData();
+    }
+
+    @Override
+    public void flatMap(NotificationOuterClass.Notification unenrichedNotification, Collector<NotificationOuterClass.Notification> collector) throws Exception {
+
+        String searchForStationId = unenrichedNotification.getDelay().getStationId().getId();
+        if (coordinateMapping.containsKey(searchForStationId)){
+            CoordinateOuterClass.Coordinate matchingCoordinate = coordinateMapping.get(searchForStationId);
+            NotificationOuterClass.DelayNotification delayNotification = unenrichedNotification.getDelay();
+            //TODO make this less ugly please!
```

Comment:
is this already less ugly if so delete this

Reply:
a cool way of  doing this would be to enrich the actual event instead of creating a new one with added information, e.g. sending it to the Builder and adding those fields. But we were not sure how to write that